File: desktop/steps/WaveformSteps.cs
using Cuke4Nuke.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Automation;
using WiPFlash;
using WiPFlash.Components;

namespace WaveformSteps
{
    public class WaveformSteps
    {
        private Application app = null;
        private Window win = null;

        [Given("^a waveform$")]
        public void AWaveform()
        {
            var launcher = new ApplicationLauncher();
            app = launcher.Recycle("WaveformApp");
            win = app.FindWindow("Waveform");

        }

        [When("^I set the Waveform Type to \"([^\"]*)\"$")]
        public void ISetTheWaveformTypeTo(string value)
        {
            win.Find<RadioButton>(value.ToLower()).Select();
        }

        [Then("^the Duty Cycle setting should be (.+)$")]
        public void TheDutyCycleSettingShouldBe(string state)
        {
            var wantEnabled = (state == "enabled");
            var dutyCycle   = win.Find<TextBox>("dutyCycle");
            var isEnabled   = (bool)dutyCycle.Element.GetCurrentPropertyValue(AutomationElement.IsEnabledProperty);

            if (isEnabled != wantEnabled)
            {
                throw new Exception("Expected duty cycle to be " + state);
            }
        }
    }
}
    

File: nohardware/features/step_definitions/waveform_steps.rb
Given(/^a waveform$/) do
  launcher = ApplicationLauncher.new
  app      = launcher.recycle 'WaveformApp'
  @win     = app.find_window  'Waveform'
end

When(/^I set the Waveform Type to "([^\"]*)"$/) do
  |value|

  radio_buttons = @win.method(:find).of(RadioButton)
  waveform_type = radio_buttons[value.downcase]
  waveform_type.select
end

Then(/^the Duty Cycle setting should be (.*)$/) do
  |state|

  text_boxes = @win.method(:find).of(TextBox)
  duty_cycle = text_boxes['dutyCycle']

  is_enabled = duty_cycle.element.
    get_current_property_value(
      AutomationElement.IsEnabledProperty)

  want_enabled = (state == "enabled")

  is_enabled.should == want_enabled
end
